src/mongodb/scores.service.ts

```diff
@@ -242,79 +247,69 @@ export class ScoresService {
     }
   }
 
+  // Target Query 
   async getTargetsBySession(sessionId: string, language: string = null) {
     const threshold = 0.9;
-    const RecordData = await this.scoreModel.aggregate([
-      {
-        $unwind: '$sessions',
-      },
+    const result = await this.scoreModel.aggregate([
       {
         $match: {
           'sessions.session_id': sessionId,
         },
       },
-      {
-        $unwind: '$sessions.confidence_scores',
-      },
-      {
-        $project: {
-          _id: 0,
-          user_id: 1,
-          date: '$sessions.date',
-          session_id: '$sessions.session_id',
-          language: '$sessions.language',
-          character: '$sessions.confidence_scores.token',
-          score: '$sessions.confidence_scores.confidence_score',
-          isRetryExists: { $ifNull: ['$sessions.isRetry', false] }
-        }
-      },
-      {
-        $match: {
-          $or: [
-            { isRetryExists: false },
-            { 'sessions.isRetry': false }
-          ]
-        }
-      },
-    ]);
-
-    const MissingRecordData = await this.scoreModel.aggregate([
       {
         $unwind: '$sessions',
       },
       {
-        $match: {
-          'sessions.session_id': sessionId,
-        },
-      },
-      {
-        $unwind: '$sessions.missing_token_scores',
-      },
-      {
-        $project: {
-          _id: 0,
-          user_id: 1,
-          date: '$sessions.date',
-          session_id: '$sessions.session_id',
-          language: '$sessions.language',
-          character: '$sessions.missing_token_scores.token',
-          score: '$sessions.missing_token_scores.confidence_score',
+        $facet: {
+          confidenceScores: [
+            {
+              $unwind: '$sessions.confidence_scores',
+            },
+            {
+              $project: {
+                _id: 0,
+                language: '$sessions.language',
+                character: '$sessions.confidence_scores.token',
+                score: '$sessions.confidence_scores.confidence_score',
+                isRetryExists: { $ifNull: ['$sessions.isRetry', false] }
+              },
+            },
+            {
+              $match: {
+                $or: [
+                  { isRetryExists: false },
+                  { 'sessions.isRetry': false }
+                ],
+              },
+            },
+          ],
+          missingTokenScores: [
+            {
+              $unwind: '$sessions.missing_token_scores',
+            },
+            {
+              $project: {
+                _id: 0,
+                language: '$sessions.language',
+                character: '$sessions.missing_token_scores.token',
+                score: '$sessions.missing_token_scores.confidence_score',
+              },
+            },
+          ],
         },
       },
     ]);
-
+    const RecordData = result[0].confidenceScores;
+    const MissingRecordData = result[0].missingTokenScores;
```

Comment:
Please implemented new logic for query which we have implemented for targets sub sessions

Reply:
Sure!